tests/Plugin_Context_Tests.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Tests for the Plugin_Context class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Plugin_Context;
+
+class Plugin_Context_Tests extends WP_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+
+		$this->plugin_context = new Plugin_Context( 'plugin-check/plugin-check.php' );
+	}
+
+	public function test_basename() {
+		$this->assertSame( 'plugin-check/plugin-check.php', $this->plugin_context->basename() );
+	}
```

Comment:
Technically, even here we could make it more reliable by using `basename( dirname( __DIR__ ) )` instead of the `plugin-check` directory. We know for sure the main file is called `plugin-check.php`, but the directory name depends on the WordPress setup and where the plugin is installed to.